lib/Environment.php

```diff
@@ -157,8 +157,10 @@ public function resolve(string $section, string $data) : ?ResolvedReference
 
             if ($this->getMetaEntry() !== null) {
                 $this->getMetaEntry()->removeDependency(
-                    // use the unique, unresolved name
-                    $this->unresolvedDependencies[$data] ?? $data
+                    // for references, use the original, unresolved name
+                    // for doc links, use the canonical URL
+                    // see addDependency()
+                    $this->unresolvedDependencies[$data] ?? (string) $this->canonicalUrl($data)
```

Comment:
The return of canonicalUrl should be handled instead of casting to a string and silencing the problem. Exception?